src/esp/gfx/replay/PlayerCallbacks.h

```diff
@@ -0,0 +1,57 @@
+#ifndef ESP_GFX_REPLAY_PLAYERCALLBACKS_H_
+#define ESP_GFX_REPLAY_PLAYERCALLBACKS_H_
+
+#include "Magnum/Magnum.h"
+
+#include <functional>
+#include <vector>
+
+namespace esp {
+namespace assets {
+struct AssetInfo;
+struct RenderAssetInstanceCreationInfo;
+}  // namespace assets
+namespace gfx {
+struct LightInfo;
+using LightSetup = std::vector<LightInfo>;
+namespace replay {
+
+// Represents either a scene::SceneNode* or batch renderer entity index.
+class GfxReplayNode;
+
+/**
+ * @brief Callbacks that contain the implementation for gfx::replay::Player.
+ */
+struct PlayerCallbacks {
+  using LoadAndCreateRenderAssetInstance = std::function<GfxReplayNode*(
```

Comment:
Why not an interface class with virtual member functions? Why all these function callbacks?

Member functions have a few benefits over callbacks that I see:
* a more common paradigm compared to passing around a pack of callbacks
* possibly lower runtime overhead (virtual function call versus `std::function`)
* elegant way to either provide default implementation or leave it undefined (pure virtual, triggering compiler error)

Reply:
An interface would make a lot more sense, I completely agree.

The idea is that, in this state, it will be easier for @mosra to integrate his changes into the replay batch renderer.

I would refactor this into an interface after the usage patterns are all present.

Reply:
As long as there is an elegant way for the callbacks to capture local variables like the resource manager instance and "this" pointer, that would work, but I am not sure there is a good way to do that. :/

Reply:
This is implemented in e2ddaaa3bae55092e7a6dcd2a2b489212ca26473, see the commit message for details.

A lambda is just a convenience syntax sugar for a stateful class, which is what I turned the implementation into. Was pretty frictionless in the end.